openupgradelib/openupgrade_merge_records.py

```diff
@@ -421,6 +425,23 @@ def _adjust_merged_values_orm(env, model_name, record_ids, target_record_id,
             if op == 'merge':
                 if not getattr(target_record, field.name) and _list:
                     vals[field.name] = _list[0]
+        elif field.type == 'many2one_reference':
+            if field.model_field not in model._fields:
+                continue
+            model_field = [x for x in fields if x.name == field.model_field][0]
+            op = op or 'merge'
+            if op == 'merge':
+                if field.model_field not in field_spec:
+                    list_model_field = all_records.mapped(field.model_field)
+                    zip_list = [(x, y) for x, y
+                                in zip(_list, list_model_field) if x and y]
+                    if not (
+                        getattr(target_record, field.name)
+                        and getattr(target_record, field.model_field)
```

Comment:
We should remove this line IMO, as we are artificially restricting the update.

Reply:
Then we need https://github.com/OCA/openupgradelib/pull/252.

Reply:
The same you could say about the `gettattr` of line https://github.com/OCA/openupgradelib/blob/master/openupgradelib/openupgrade_merge_records.py#L422, that also could be removed, right?

Reply:
It's not the same. I'm talking about removing this specific line, not the previous one. The first check is for not overwriting if there's already a value set. This second one is for forcing to have a value in the other field.

Reply:
Anyways, this is not critical of course. Both fields should be filled in normal conditions and not filled on contrary. To have one filled and the other not is the anomaly.

Reply:
Imagine you have target record with m2o_r = 0, model = 'ir.attachment', record 2 with m2o_r = 0, model = 'ir.attachment', and record 3 with m2o_r = 1, model = 'ir.attachment'. You want target_record having m2o_r = 1, model = 'ir.attachment'. That's why we need double check.

Reply:
Yeah, you are right. Done.